animation_nodes/nodes/rotation/c_utils.pyx

```diff
@@ -100,3 +101,22 @@ def getAxisListOfQuaternionList(QuaternionList quaternions, str axis):
         for i in range(output.length):
             output.data[i] = quaternions.data[i].z
     return output
+
+#base on the expression from http://planning.cs.uiuc.edu/node198.html?
+def randomQuaternionList(int seed, int amount):
+    cdef QuaternionList result = QuaternionList(length = amount)
+    cdef XoShiRo256Plus rng = XoShiRo256Plus(seed)
+    cdef double u1,u2,u3
+    cdef Py_ssize_t i
+    for i in range(amount):
+        u1 = rng.nextFloat()
+        u2 = rng.nextFloat()
+        u3 = rng.nextFloat()
+        
+        result.data[i].w = sqrt(1-u1) * sin(2*PI*u2)
+        result.data[i].x = sqrt(1-u1) * cos(2*PI*u2)
+        result.data[i].y = sqrt(u1) * sin(2*PI*u3)
+        result.data[i].z = sqrt(u1) * cos(2*PI*u3)
```

Comment:
You should group those expressions. For instance, `u2 = rng.nextFloat() * 2 * PI`. Also, store `sqrt(1 - u1)` in a variable not to compute it twice. Finally, make sure to put spaces around operators (`x+y` -> `x + y`).